src/Panel/TimeLayerSliderPanel/TimeLayerSliderPanel.tsx

```diff
@@ -86,318 +77,254 @@ export class TimeLayerSliderPanel extends React.Component<TimeLayerSliderPanelPr
     },
-    autoPlaySpeedOptions: [ 0.5, 1, 2, 5, 10, 100, 300 ]
-  };
-
-  private _wmsTimeLayers: TimeLayerAwareConfig[] = [];
-  private _interval: number;
-
-  /**
-   * Constructs time panel.
-   */
-  constructor(props: TimeLayerSliderPanelProps) {
-    super(props);
-
-    this.state = {
-      value: moment().milliseconds(0),
-      playbackSpeed: 1,
-      autoPlayActive: false,
-      startDate: moment().milliseconds(0),
-      endDate: moment().milliseconds(0),
-    };
-    this._interval = 1000;
-
-    this.wrapTimeSlider();
-
-    // binds
-    this.onTimeChanged = this.onTimeChanged.bind(this);
-    this.autoPlay = this.autoPlay.bind(this);
-    this.updateDataRange = this.updateDataRange.bind(this);
-  }
-
-  componentDidMount() {
-    const {
-      initStartDate,
-      initEndDate
-    } = this.props;
-
-    this.setState({
-      startDate: initStartDate,
-      endDate: initEndDate
-    });
-  }
-
-  componentDidUpdate(prevProps: TimeLayerSliderPanelProps) {
-    prevProps.timeAwareLayers.forEach((pl, i: number) => {
-      const tpl = this.props.timeAwareLayers[i];
-      if (!(_isEqual(getUid(pl), getUid(tpl)))) {
-        // update slider properties if layers were updated
-        this.wrapTimeSlider();
-        this.findRangeForLayers();
-      }
-    });
-  }
-
-  /**
-   *
-   * @param nextProps
-   * @param nextState
-   */
-  shouldComponentUpdate(nextProps: TimeLayerSliderPanelProps, nextState: TimeLayerSliderPanelState) {
-    const {
-      value,
-      autoPlayActive,
-      startDate,
-      endDate,
-    } = this.state;
-    const {
-      timeAwareLayers,
-      tooltips
-    } = this.props;
-
-    if (nextState.value !== value) {
-      return true;
-    }
-    if (nextState.autoPlayActive !== autoPlayActive) {
-      return true;
-    }
-    if (nextState.startDate !== startDate) {
-      return true;
-    }
-    if (nextState.endDate !== endDate) {
-      return true;
-    }
-    if (!(_isEqual(nextProps.timeAwareLayers, timeAwareLayers))) {
-      return true;
-    }
-
-    return !(_isEqual(nextProps.tooltips, tooltips));
-  }
+    autoPlaySpeedOptions = [0.5, 1, 2, 5, 10, 100, 300],
+    initStartDate = moment(moment.now()),
+    initEndDate = moment(moment.now()).add(1, 'days')
+  }) => {
+    const [currentValue, setCurrentValue] = useState<Moment>(value);
+    const [playbackSpeed, setPlaybackSpeed] = useState<
+      number | PlaybackSpeedType
+    >(1);
+    const [autoPlayActive, setAutoPlayActive] = useState(false);
+    const [startDate, setStartDate] = useState<Moment>(initStartDate);
+    const [endDate, setEndDate] = useState<Moment>(initEndDate);
+
+    const wmsTimeLayersRef = useRef<TimeLayerAwareConfig[]>([]);
+    const intervalRef = useRef<number | undefined>(1000);
+    const prevPropsRef = useRef<TimeLayerSliderPanelProps>();
+
+    const wrapTimeSlider = useCallback(() => {
+      const wmsTimeLayers: TimeLayerAwareConfig[] = [];
+      timeAwareLayers.forEach((l: any) => {
```

Comment:
is it possible to use OlLayer here instead of any?

Reply:
Yes, changed the typing here.